src/classes/HH_Container_LCTRL.cls

```diff
@@ -271,13 +272,24 @@ public with sharing class HH_Container_LCTRL {
             'MailingCountry', 
             'MailingLatitude', 
             'MailingLongitude'};
-        if (UTIL_Describe.isMiddleNameEnabled()) {
-            setContactFields.add('Middlename');
-        }
+
+        addContactMiddleNameField(setContactFields);
+
         return setContactFields;
     }
 
     /*******************************************************************************************************
+    * @description Adds MiddleName field into the set of field names
+    * @param fieldNames Set of field names 
+    * @return void
+    */
+    private static void addContactMiddleNameField(Set<String> fieldNames) {        
+        if (UTIL_Describe.isMiddleNameEnabled()) {
+            fieldNames.add('MiddleName');
```

Comment:
Good question @force2b . I did consider case insensitive field names and added a method to process them as such, so Manage Household page does not generate an exception (difference can be seen when dev is compared with the first commit in this branch). However, this is not the only place when HH Naming configuration fields are used, so the exception is not local to the Managed Household page in the case configuration is not as expected. There is a backlog ticket that will ensure configuration is properly setup since the exception will be raised for any field (using firstname instead of FirstName for example). Therefore, the code to check for case insensitive names in Manage Household becomes obsolete if it is expected that field names will be as expected, ie case sensitive API names. Thus, the changes are made in the second commit of this branch to take care of the MiddleName issue for which the bug has been created.